### PR TITLE
Java has simple execution support

### PR DESCRIPTION
**Pull request by gogo4ds**
Add judge support for Java simple execution strategy
https://github.com/SoftUni-Internal/interactive-issues/issues/247

### CODE REVIEW

File: OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public class JavaPreprocessCompileExecuteAndCheckExecutionStrategy : BaseCompile
         protected Func<CompilerType, string> GetCompilerPathFunc { get; }
 
         protected string SandboxExecutorSourceFilePath =>
```
Comment (achobanov) [inline OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs]: Arrow bellow.

File: OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs [hunk 2]
```diff
@@ -45,7 +45,7 @@ public class JavaPreprocessCompileExecuteAndCheckExecutionStrategy : BaseCompile
...
-            $"{this.WorkingDirectory}\\{SandboxExecutorClassName}{Constants.JavaSourceFileExtension}";
+            $"{Path.Combine(this.WorkingDirectory, SandboxExecutorClassName)}{Constants.JavaSourceFileExtension}";
 
         protected string SandboxExecutorCode => @"
```
Comment (achobanov) [inline OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs]: Arrow bellow.

File: OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs [hunk 3]
```diff
@@ -277,7 +235,31 @@ class _$SandboxSecurityManager extends SecurityManager {
...
+                executionContext,
+                compileResult.OutputFile,
+                executionContext.Input);
+
+            result.Results.Add(this.GetOutputResult(processExecutionResult));
+
+            return result;
+        }
+
+        protected virtual string CreateSubmissionFile<TInput>(IExecutionContext<TInput> executionContext) =>
```
Comment (achobanov) [inline OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs]: Put parameter and arrow on new lines.

File: OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs [hunk 4]
```diff
@@ -277,7 +235,31 @@ class _$SandboxSecurityManager extends SecurityManager {
...
             JavaCodePreprocessorHelper.CreateSubmissionFile(executionContext.Code, this.WorkingDirectory);
```
Comment (achobanov) [inline OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs]: Parameters on new line.

File: OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs [hunk 5]
```diff
@@ -296,6 +278,73 @@ class _$SandboxSecurityManager extends SecurityManager {
...
+                .Replace('\\', '.');
+
+            var timeMeasurementFilePath = $"{this.WorkingDirectory}\\{TimeMeasurementFileName}";
+
+            var processExecutionResult = executor.Execute(
+                    this.JavaExecutablePath,
+                    input,
+                    executionContext.TimeLimit * 2, // Java virtual machine takes more time to start up
+                    executionContext.MemoryLimit,
+                    new[] { classPathArgument, SandboxExecutorClassName, classToExecute, $"\"{timeMeasurementFilePath}\"" },
```
Comment (achobanov) [inline OJS.Workers.ExecutionStrategies/Java/JavaPreprocessCompileExecuteAndCheckExecutionStrategy.cs]: Span array declaration on multiple lines.

File: OJS.Workers.ExecutionStrategies/Java/JavaUnitTestsExecutionStrategy.cs [hunk 6]
```diff
@@ -262,7 +262,7 @@ public class _$TestRunner {{
             return compilerResult;
         }
 
-        protected override string CreateSubmissionFile(IExecutionContext<TestsInputModel> executionContext)
+        protected override string CreateSubmissionFile<TInput>(IExecutionContext<TInput> executionContext)
```
Comment (achobanov) [inline OJS.Workers.ExecutionStrategies/Java/JavaUnitTestsExecutionStrategy.cs]: Parameter on new line.

File: OJS.Workers.ExecutionStrategies/Java/JavaZipFileCompileExecuteAndCheckExecutionStrategy.cs [hunk 7]
```diff
@@ -25,7 +25,7 @@ public class JavaZipFileCompileExecuteAndCheckExecutionStrategy : JavaPreprocess
         {
         }
 
-        protected override string CreateSubmissionFile(IExecutionContext<TestsInputModel> executionContext)
+        protected override string CreateSubmissionFile<TInput>(IExecutionContext<TInput> executionContext)
```
Comment (achobanov) [inline OJS.Workers.ExecutionStrategies/Java/JavaZipFileCompileExecuteAndCheckExecutionStrategy.cs]: Parameter on new line.

File: OJS.Workers.ExecutionStrategies/Java/JavaUnitTestsExecutionStrategy.cs (hunk 6 above)
Comment (gogo4ds) [reply OJS.Workers.ExecutionStrategies/Java/JavaUnitTestsExecutionStrategy.cs]: No need. It is less than 120 characters

File: OJS.Workers.ExecutionStrategies/Java/JavaZipFileCompileExecuteAndCheckExecutionStrategy.cs (hunk 7 above)
Comment (gogo4ds) [reply OJS.Workers.ExecutionStrategies/Java/JavaZipFileCompileExecuteAndCheckExecutionStrategy.cs]: No need. It is less than 120 characters

Comment (gogo4ds) [reply OJS.Workers.ExecutionStrategies/Java/JavaZipFileCompileExecuteAndCheckExecutionStrategy.cs]: No need. It is less than 120 characters

**Pull request merged**